### PR TITLE
TypeError: Cannot read property 'replace' of undefined

### PR DESCRIPTION
**Pull request by ChocPanda**
I am getting the following error in my CI pipeline. I'm not yet sure of the cause but unfortunately the below error is preventing me from creating a proper test report.

```
/home/circleci/project/node_modules/trim/index.js:5
  return str.replace(/^\s*|\s*$/g, '');
             ^

TypeError: Cannot read property 'replace' of undefined
    at trim (/home/circleci/project/node_modules/trim/index.js:5:14)
    at Parser._handleError (/home/circleci/project/node_modules/tap-out/index.js:189:14)
    at Parser.handleLine (/home/circleci/project/node_modules/tap-out/index.js:48:8)
    at Stream.<anonymous> (/home/circleci/project/node_modules/tap-out/index.js:256:14)
    at Stream.emit (events.js:200:13)
    at drain (/home/circleci/project/node_modules/through/index.js:36:16)
    at Stream.stream.queue.stream.push (/home/circleci/project/node_modules/through/index.js:45:5)
    at emit (/home/circleci/project/node_modules/split/index.js:37:14)
    at next (/home/circleci/project/node_modules/split/index.js:49:7)
    at Stream.<anonymous> (/home/circleci/project/node_modules/split/index.js:54:5)
```

I saw that in line 151 of the index file you account for the possibility that `m` may only have a single member so I have simply repeated the logic below in line 180 to prevent the error.

I also noted that you appeared to be using a rather old library for trimming your strings, as this functionality is now a member of the String type I took the liberty of updating that as well, I hope this is okay

### CODE REVIEW

**Comment by ChocPanda** [comment]
@scottcorgan Any chance you could take a look at this?

**Comment by 3cp** [comment]
@scottcorgan please have a look of this fix.

**Comment by ChocPanda** [comment]
@scottcorgan Can you please review this

**Comment by Sceat** [comment]
@ChocPanda this is bugging me since quite some time, can you publish your fork to npm, i will use it in https://github.com/Sceat/tap-spec-emoji

**Comment by ChocPanda** [comment]
Sure, I'll post a link here when I'm done

File: index.js [hunk 1]
```diff
@@ -177,16 +176,16 @@ Parser.prototype._handleError = function _handleError(line) {
 
       // Need to set this value
       if (m[0] === 'actual') {
-        lastAssert.error.actual = trim(m[1]);
+        lastAssert.error.actual = m[1].trim();
```
Comment (Sceat) [inline index.js:179]: maybe we should make sure the values are always there before calling trim ? the following apply a bit everywhere (yet it require a recent node version so we should write it on the readme)
```suggestion
        lastAssert.error.actual = m[1]?.trim();
```

File: index.js [hunk 2]
```diff
@@ -177,16 +176,16 @@ Parser.prototype._handleError = function _handleError(line) {
...
     }
 
     // outputting expected/actual object or array
     if (this.currentNextLineError) {
-      lastAssert.error[this.currentNextLineError] = trim(line);
+      lastAssert.error[this.currentNextLineError] = line.trim();
       this.currentNextLineError = null;
     }
-    else if (trim(m[1]) === '|-') {
+    else if ((m[1] || '').trim() === '|-') {
```
Comment (Sceat) [inline index.js:188]: like those cases can be replaced
```suggestion
    else if (m[1]?.trim() === '|-') {
```

File: index.js (hunk 1 above)
Comment (ChocPanda) [reply index.js:179]: I think publishing for more recent versions of node would be a good idea. We could add a build and packaging step to transpile for backwards compatibility. I have published what I have for now and will look to maintain and update in the future. Feel free to contribute to the fork and I'll try to keep up

**Comment by ChocPanda** [comment]
@Sceat @3cp  I have published my fork here [to NPM here](https://www.npmjs.com/package/@chocpanda/tap-out)

**Comment by Sceat** [comment]
now using it in https://github.com/Sceat/tap-spec-emoji

**Comment by Sceat** [comment]
working good with the `at` in [@hydre/doubt](https://github.com/HydreIO/doubt) :+1: 

image

**Pull request closed without merging**